src/common/post.ts

```diff
@@ -252,3 +257,68 @@ export const validatePost = (
 export const submitArticleThreshold = parseInt(
   process.env.SUBMIT_ARTICLE_THRESHOLD,
 );
+
+export const insertCodeSnippets = async ({
+  entityManager,
+  post,
+  codeSnippetsUrl,
+}: {
+  entityManager: EntityManager;
+  post: Pick<Post, 'id'>;
+  codeSnippetsUrl: string | undefined;
+}) => {
+  if (!codeSnippetsUrl) {
+    return;
+  }
+
+  try {
+    const codeSnippetsJson = await downloadJsonFile<PostCodeSnippetJsonFile>({
+      url: codeSnippetsUrl,
+    });
+
+    const codeSnippets = codeSnippetsJson.snippets.reduce(
+      (acc, codeSnippetContent, index) => {
+        const checksum = createHash('sha1');
+        checksum.update(codeSnippetContent);
+        const contentHash = checksum.digest('hex');
+
+        if (!acc.has(contentHash)) {
+          acc.set(
+            contentHash,
+            entityManager.getRepository(PostCodeSnippet).create({
+              postId: post.id,
+              contentHash,
+              order: index,
+              content: codeSnippetContent,
+            }),
+          );
+        }
+
+        return acc;
+      },
+      new Map<string, PostCodeSnippet>(),
+    );
+
+    const uniqueCodeSnippets = Array.from(codeSnippets.values());
+
+    await entityManager.getRepository(PostCodeSnippet).delete({
+      postId: post.id,
+      contentHash: Not(
+        In(uniqueCodeSnippets.map((snippet) => snippet.contentHash)),
+      ),
+    });
+
+    await entityManager
+      .getRepository(PostCodeSnippet)
+      .upsert(uniqueCodeSnippets, {
+        conflictPaths: ['postId', 'contentHash'],
+      });
```

Comment:
We always cleanup snippets that are no longer present in article and then upsert the new ones.